server/src/WebHook.hs

```diff
@@ -135,6 +149,45 @@ runWebHook uuid logger Config{..} commit =
             defaultLogLevel
         $ do
             postStatus' Pending "Working on checks..."
-            postStatus' Success "No checks implemented yet"
+            handleCompileResult uuid commit config
+            handleTestingResult uuid commit config
+  where
+    postStatus' = postStatus uuid githubToken domain commit
+
+handleCompileResult :: UUID -> Commit -> Config -> LogT IO ()
+handleCompileResult uuid commit Config{..} = do
+    postStatus' Pending "Compilation in progress..."
+    compileResult <- liftIO $ invokeMavenCommand "compile"
+    case compileResult of
+        Left errMsg -> do
+            logAttention_ $ format "Failed to compile: {}" errMsg
+            postStatus' Error "Compilation failed!"
+        Right (exitCode, compileOutput) -> do
+            currentTime <- liftIO getCurrentTime
+            liftIO $
+                runSql $
+                    insertBuild $
+                        Build
+                            (toText uuid)
+                            (after commit)
+                            (pack $ formatTime defaultTimeLocale "%F %T %z" currentTime)
+                            (pack compileOutput)
+            logInfo_ $ format "Compilation succeeded with exit code: {}" exitCode
+            postStatus' Success "Compilation succeeded!"
+  where
+    postStatus' = postStatus uuid githubToken domain commit
+    runSql = runSqlite $ pack dbFile
+
+handleTestingResult :: UUID -> Commit -> Config -> LogT IO ()
```

Comment:
This looks very similar to `handleCompileResult`, maybe it is a good idea to just use one function? I can think of something like:
```haskell
runMaven :: UUID -> Commit -> Config -> Text -> (ExitCode, Text)
```
The test log should be put into database as well.